checker/src/utils.rs

```diff
@@ -58,6 +59,139 @@ pub fn is_public(def_id: DefId, tcx: &TyCtxt<'_, '_, '_>) -> bool {
     }
 }
 
+/// Returns a string that is a valid identifier, made up from the concatenation of
+/// the string representationss of the given list of argument types.
+pub fn argument_types_key_str<'tcx>(tcx: &TyCtxt<'_, '_, 'tcx>, ty: Ty<'tcx>) -> String {
+    let mut result = String::new();
+    let bound_sig = ty.fn_sig(*tcx);
+    let sig = bound_sig.skip_binder();
+    for arg_ty in sig.inputs().iter() {
+        result.push('_');
```

Comment:
_ isn't a valid part of the mangled type, I'm guessing?

Reply:
I'm none to sure what you are asking here.

Reply:
Whoops, that was half a thought that I re-worded in argument_types_key_str. Ignore.